### ISSUE TITLE
No way to batch builds on a per branch basis

### ISSUE THREAD

**Issue by dalemyers** [Microsoft/azure-pipelines-agent#1830]
## What's not working?

Before we moved to YAML, we had half a dozen build definitions so that we could set different variables. Moving to YAML allowed us to consolidate these, with the exception of one variable: We cannot batch changes on a per branch basis any longer. 

Our definition builds all branches (and is a policy for PRs). We'd like 3 of our branches to have their changes batched (on a per branch basis), but all the rest continue as is. 

Currently, I cannot see any way to do that. Issue #1549 brings the first part, but without this remaining part, we are likely going to have to move back to multiple definitions anyway.

**Comment by TingluoHuang** [comment]
@chrisrpatterson @ericsciple

**Comment by ericsciple** [comment]
@dalemyers try [batch: true](https://github.com/Microsoft/azure-pipelines-agent/blob/3d39fca3cac2bf54c16600e0a3b9e7972d4c5ef3/docs/preview/yamlgettingstarted-ci.md)

**Comment by dalemyers** [comment]
That sets it on a per definition basis, not a per branch basis.

**Comment by ericsciple** [comment]
Batching is applied per branch. If not, that's a bug. Did you experience that behavior? If so, I'll see if I can reproduce the behavior you are describing.

**Comment by dalemyers** [comment]
Yup. All our builds stopped bar one. I undid the change immediately without investigation though since obviously I don't want to kill our builds.

**Comment by ericsciple** [comment]
I am not able to reproduce this behavior.

Here is the definition I used to test this on my personal account:

```yaml
trigger:
  batch: true
  branches:
    include:
    - "*"
pool: hosted ubuntu 1604
steps:
- script: ping -c 600 127.0.0.1
```

**Comment by ericsciple** [comment]
I wonder if what you experienced was a delay due to some other reason or if the UI was not up to date. I'm going to close this issue here for now since this feature appears to work correctly (or at least now it does). Please reopen if you are still having issues.

**Comment by dalemyers** [comment]
I tried this again today to see if it would work. Here's what I did:

1. I added a commit to the `develop` branch which contains the following added to the root level of the YAML file:
```
trigger:
  batch: true
```
2. I then pushed this commit to the repo on VSTS. The commit did not build. It was never scheduled for building. I checked all our agents and none of them were showing as having an active build. There were no status updates on the commit as we'd expect if it was starting to build. 

3. I created a new branch based on this commit, made a change, commited and pushed. This branch also did not build. Adding another commit to this branch resulted in no build either. 

I did not test what happened if I pushed a commit based on an earlier revision of `develop` as I needed to get the change back out. I also did not wait for existing builds to finish to see if one of the commits I pushed would get a build scheduled as I know from our previous test that this is in fact what happens. 

It seems that this just isn't working for us and I have no idea why.
